test/cypress/integration/group-chat.spec.js

```diff
@@ -498,7 +498,7 @@ describe('Group Chat Basic Features (Create & Join & Leave & Close)', () => {
     cy.giLogout()
   })
 
-  it(`user2 joins the ${groupName1} group and ${CHATROOM_GENERAL_NAME} again and logout`, () => {
+  it(`user2 joins the ${groupName1} group and ${CHATROOM_GENERAL_NAME} channel again and logout`, () => {
```

Comment:
On my local test run the last test in `group-chat.spec.js` (`user2 joins the ${groupName1} group and ${CHATROOM_GENERAL_NAME} channel again and logout`) failed:

https://github.com/user-attachments/assets/136109de-d25c-4cc1-8ae8-08cb0b74f2b5

This is failing here inside of `cy.giAcceptGroupInvite`:

```js
  if (existingMemberUsername) {
    // NOTE: checking 'data-groupId' is for waiting until joining process would be finished
    cy.getByDT('pendingApprovalTitle').invoke('attr', 'data-groupId').should('eq', groupId)
    // NOTE: should wait until KEY_REQUEST event is published
    cy.giKeyRequestedGroupIDs(groupId)
    cy.giEmptyInvocationQueue() // <---------remove

    cy.giLogout()
```

And then in:

```js
Cypress.Commands.add('giLogout', ({ hasNoGroup = false } = {}) => {
  cy.giEmptyInvocationQueue()

  if (hasNoGroup) {
    cy.window().its('sbp').then(async sbp => await sbp('gi.app/identity/logout'))
  } else {
    cy.getByDT('settingsBtn').click() // <------ failing here
```

There are multiple problems here:

1. `cy.giEmptyInvocationQueue()` is being called an extra time before `cy.giLogout()` (which also calls it) - so that unnecessary call in `giAcceptGroupInvite` on line `commands.js:498` should be removed.
2. And the main problem: for some reason we were given the key to join the group and so the approval screen is visible here and the settings button to log out cannot be clicked, and also there is no need for it be clicked because we're already in the group

It might be because `user2` was already a member of the group before, and so was somehow able join again because the keys were not properly rotated when he left in `'user2 leaves the group by himself'`.

However, running the tests a second time everything passed, so this is a challenging and important Heisenbug that needs to be fixed. I will open an issue for this.

cc @corrideat

Reply:
Yeah, that makes sense. I remember I have already encountered the second issue several times before.
Since you created an issue for this heisenbug, we can check it out and work on it later.

Reply:
I'd like to leave this unresolved just so that when the issue links to it this conversation can be seen.